app/models/agents/post_agent.rb

```diff
@@ -66,10 +75,19 @@ def validate_options
         errors.add(:base, "post_url and expected_receive_period_in_days are required fields")
       end
 
-      if options['payload'].present? && !options['payload'].is_a?(Hash)
+      if options['payload'].present? && %w[get delete].include?(method) && !options['payload'].is_a?(Hash)
         errors.add(:base, "if provided, payload must be a hash")
       end
 
+      if options['payload'].present? && %w[post put patch].include?(method)
+        if !(options['content_type'] =~ MIME_RE || options['payload'].is_a?(Hash))
```

Comment:
Is this more readable? 

`if !options['payload'].is_a?(Hash) && options['content_type'] !~ MIME_RE`